### ISSUE TITLE
VT is Heating even if Target Temperature is reached and does not stop

### ISSUE THREAD

**Issue by Caliban2017** [jmcollin78/versatile_thermostat#649]
Hello, i installed Versatile Thermostat and use 5 Devices successfully with it, but one of them is making some Trouble. It is constantly heating, even when the Target Temperature is set to 15°C, it is reaching 21° and above and does not stop Heating. It seems like the VT is asking for the Heat, as the Screenshot will tell here:

![therm1](https://github.com/user-attachments/assets/f5a23326-d687-4494-89c7-f9ff4a3eef96)

All other Thermostats of the same Type work fine, a example of the same Thermostat in another Room:

![therm2](https://github.com/user-attachments/assets/650d7460-9bb0-4458-8457-4307ef2eeb18)

What can cause this Problem?? I already deleted the VT, created it new but always the same... It completely overshoots the Target Temperature and does not stop to heat, it still continues....

Here are Config of the Thermostat:

hvac_modes: heat, auto, off
min_temp: 7
max_temp: 35
target_temp_step: 0.1
preset_modes: none, frost, eco, comfort, boost
current_temperature: 19.8
temperature: 15
hvac_action: idle
preset_mode: eco
is_on: false
hvac_mode: off
type: null
is_controlled_by_central_mode: true
last_central_mode: Auto
frost_temp: 10
eco_temp: 15
boost_temp: 25
comfort_temp: 20
frost_away_temp: 0
eco_away_temp: 0
boost_away_temp: 0
comfort_away_temp: 0
power_temp: null
target_temperature_step: 0.1
ext_current_temperature: 5.62
ac_mode: false
current_power: null
current_power_max: null
saved_preset_mode: eco
saved_target_temp: 15
saved_hvac_mode: null
motion_sensor_entity_id: null
motion_state: null
power_sensor_entity_id: null
max_power_sensor_entity_id: null
overpowering_state: null
presence_sensor_entity_id: null
presence_state: null
window_state: off
window_auto_state: off
window_bypass_state: false
window_sensor_entity_id: null
window_delay_sec: null
window_auto_enabled: false
window_auto_open_threshold: null
window_auto_close_threshold: null
window_auto_max_duration: null
window_action: window_turn_off
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: 2024-11-17T01:01:38.285998+01:00
last_ext_temperature_datetime: 2024-11-17T01:02:06.987365+01:00
security_state: false
minimal_activation_delay_sec: 10
device_power: 1
mean_cycle_power: null
total_energy: 32.89
last_update_datetime: 2024-11-17T01:02:06.990388+01:00
timezone: Europe/Berlin
temperature_unit: °C
is_device_active: false
ema_temp: 19.79
is_used_by_central_boiler: false
temperature_slope: -0.31
hvac_off_reason: null
is_over_climate: true
start_hvac_action_date: null
underlying_entities: climate.thermostat_bad
is_regulated: true
regulated_target_temperature: 14
auto_regulation_mode: auto_regulation_medium
regulation_accumulated_error: -20
auto_fan_mode: auto_fan_high
current_auto_fan_mode: auto_fan_high
auto_activated_fan_mode: null
auto_deactivated_fan_mode: null
auto_regulation_use_device_temp: false
auto_start_stop_enable: false
auto_start_stop_level: auto_start_stop_none
auto_start_stop_dtmin: null
auto_start_stop_accumulated_error: 0
auto_start_stop_accumulated_error_threshold: null
follow_underlying_temp_change: false
friendly_name: BT - Bad
supported_features: 401

**Comment by kvanbiesen** [comment]
Same happened here somewhere in the last updates. Can't pinpoint it but seems to have happen after the auto start/stop function was added.

Updating the preset from the thermostat immediately stops the heating mode

**Comment by Caliban2017** [comment]
I am not sure if i found the issue, but maybe i did ... i removed the whole Thermostat again, added in again in VT but gave it another Name and i noticed, in Zigbee2MQTT, there is a Setting where i can choose the Preset. All other Thermostats which are working where on "manual" Preset, i think because VT is taking over the Control here and no internal Preset is needed (?!). On the Bathroom Thermostat whoever the Preset was set to "comfort", so it tried to reach 21°C all the time. I put it now manually to the "manual" Preset and readded the Thermostat like i said with another Name, to be sure it does not use some old Config. For now, it did not heat up again so far, i hope it works now again.

But i can confirm what @kvanbiesen said, the Preset Behaviour on the UI Card is very clunky, after a Restart of Home Assistant, all Settings seem to be lost and will not be restored, some Thermostats which where On turn Off after a restart, some lose the selected Presets, so they are off and there is so Preset selected, very strange. Same with the Scheduler Component, i use it to time the Thermostats over the Day, but sometimes the Presets will not be accepted and when i look at the Dashboard, there is just no Preset selected (all Icons are greyed out).

**Comment by jmcollin78** [comment]
> Can't pinpoint it but seems to have happen after the auto start/stop function was added.

Auto-start/stop doesn't change regulation algorithm. So, I don't see the connection.
What type of VTherm do you have (over_climate like @Caliban2017 ?) ? What are the attributes ? Give me some curve that shows the problem.

**Comment by kvanbiesen** [comment]
Wait ..I found what causes mine....
It's nothing wrong with the regulation. It's something to do with the manual change in latest update.

Since version 6.7. vtherm doesn't defect it anymore when a "over_climate"  get changed by hand. It would set preset to none/manual. Now if you flick the knob, the preset stays stuck in the current temperature. 

So vtherm is still targeting the preset temperature but the trv us using the internal temperature.
Before 6.7 I believe the changes on the trv itself were seen by vtherm. Have to roll back to see in which version it changed.

**Comment by jmcollin78** [comment]
> Wait ..I found what causes mine.... It's nothing wrong with the regulation. It's something to do with the manual change in latest update.
> 
> Since version 6.7. vtherm doesn't defect it anymore when a "over_climate" get changed by hand. It would set preset to none/manual. Now if you flick the knob, the preset stays stuck in the current temperature.
> 
> So vtherm is still targeting the preset temperature but the trv us using the internal temperature. Before 6.7 I believe the changes on the trv itself were seen by vtherm. Have to roll back to see in which version it changed.

If you enable the "Follow the under...", the behavior will be exactly as previous release.

**Comment by Johnson145** [comment]
I have a similar problem, but I'm using the `thermostat_over_valve` type. The target temperature is already reached by a few degrees, but VT keeps the valve opening at 100% and heats way too much.

For that thermostat type, there is no `Follow the under...`, right?

I also can't change the preset anymore. The preset icons in the UI remain greyed out. 

The logs don't show any errors or warnings.

At some point I installed the latest Home Assistant (Core?) Update and restarted the instance. Afterwards, VT stopped heating and I could select a preset again. Anyway, I've "played around a little with different temperature and preset settings" and I'm back to the original problem: VT keeps on heating at 100% even though the target temperature has already been reached.

A while after, I noticed the opposite effect: I increased the target temperature above the current one, but VT does not start to heat.

I I wait long enough, nothing changes automatically. Anyway, if I increase the temperature even further, it suddenly updates the heating status. If I decrease the target temperature again below the current temperature, it again keeps on heating instead of stopping.

Is there maybe some internal minimum time span that needs to pass between two consecutive temperature configuration changes?

Not sure while the additional symptoms mean that I have a different problem and shouldn't bother you in this issue.
